RecoMTD/TrackExtender/plugins/TrackExtenderWithMTD.cc

```diff
@@ -146,15 +207,54 @@ namespace {
 
     tofpid.gammasq_pi = 1. + magp2 * m_pi_inv2;
     tofpid.beta_pi = std::sqrt(1. - 1. / tofpid.gammasq_pi);
-    tofpid.dt_pi = tofpid.pathlength / tofpid.beta_pi * c_inv;
+    switch (choice) {
+      case TofCalc::cost:
+        tofpid.dt_pi = tofpid.pathlength / tofpid.beta_pi * c_inv;
+        break;
+      case TofCalc::segm:
+        tofpid.dt_pi = trs.computeTof(m_pi_inv2);
+        break;
+      case TofCalc::mixd:
+        tofpid.dt_pi = trs.computeTof(m_pi_inv2) + tofpid.pathlength / tofpid.beta_pi * c_inv;
+        break;
+      default:
+        tofpid.dt_pi = 1;
+        break;
+    }
 
     tofpid.gammasq_k = 1. + magp2 * m_k_inv2;
     tofpid.beta_k = std::sqrt(1. - 1. / tofpid.gammasq_k);
-    tofpid.dt_k = tofpid.pathlength / tofpid.beta_k * c_inv;
+    switch (choice) {
+      case TofCalc::cost:
+        tofpid.dt_k = tofpid.pathlength / tofpid.beta_k * c_inv;
+        break;
+      case TofCalc::segm:
+        tofpid.dt_k = trs.computeTof(m_k_inv2);
+        break;
+      case TofCalc::mixd:
+        tofpid.dt_k = trs.computeTof(m_k_inv2) + tofpid.pathlength / tofpid.beta_k * c_inv;
+        break;
+      default:
+        tofpid.dt_k = 1;
+        break;
+    }
```

Comment:
this and the proton part is almost a full repeat of the `pi` case; the implementation is now long enough that a refactoring to a single method seems necessary to compute the necessary `dt`

Reply:
not really necessary, but I agree aesthetically more appealing. A lambda function should be an effective solution